fs2/shared/src/main/scala/org/atnos/eff/addon/fs2/TaskEffect.scala

```diff
@@ -0,0 +1,229 @@
+package org.atnos.eff.addon.fs2
+
+import cats._
+import cats.implicits._
+import fs2._
+import org.atnos.eff._
+import org.atnos.eff.syntax.all._
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.TimeoutException
+import scala.util._
+
+case class TimedTask[A](task: (Strategy, Scheduler) => Task[A], timeout: Option[FiniteDuration] = None) {
+  def runNow(implicit strategy: Strategy, scheduler: Scheduler): Task[A] = timeout.fold(task(strategy, scheduler)) { t =>
+    for {
+      ref <- Task.ref[A]
+      _ <- ref.set(
+        (Task.fail(new TimeoutException).schedule(t): Task[A])
+          .race(task(strategy, scheduler))
+          .map(_.merge)
+      )
+      result <- ref.get
+    } yield result
+  }
+}
+
+object TimedTask {
+  final def TimedTaskApplicative: Applicative[TimedTask] = new Applicative[TimedTask] {
+    override def pure[A](x: A) = TimedTask((_, _) => Task.now(x))
```

Comment:
Oh, I see the pattern. You only use `override` on methods that aren't required to be implemented. I'll fix mine.